python/tvm/topi/nn/dense.py

```diff
@@ -94,9 +94,9 @@ def matmul(
         red_dim, out_dim = tensor_b.shape
 
     # cmp should be done by values
-    assert int(in_dim) == int(
-        red_dim
-    ), "Inner dimensions of dense do not match. {in_dim} vs {red_dim}."
+    if isinstance(in_dim, tvm.tir.expr.IntImm) and isinstance(red_dim, tvm.tir.expr.IntImm):
+        assert int(in_dim) == int(red_dim), \
+            "Inner dimensions of dense do not match. {in_dim} vs {red_dim}."
```

Comment:
we need to think about the correct way to handle these sort of (non-additive) changes in our workflow. ideally they'd go straight to apache/tvm but we'd need to find a way to test the change there.